docs/domain-driven-design.md

```diff
@@ -0,0 +1,118 @@
+# Domain Driven Design
+
+The codebase and project attempts to follow the principles of Domain Driven Design (DDD) as much as possible, and
+wherever it makes sense. DDD is a software development approach that focuses on the domain, the problem that the
+software is trying to solve, and the business logic that drives the problem.
+
+It is a way of thinking and a set of priorities, aimed at accelerating software projects that have to deal with
+complex domains that are constantly changing.
+
+- [Principles of DDD]
```

Comment:
Should these be links to the respective headers?

Reply:
In that case:
- [Principles of DDD](#principles-of-ddd)
- [Hexagonal Architecture](#hexagonal-architecture)
- [Implemented in code](#implemented-in-code)

Reply:
Yep my bad